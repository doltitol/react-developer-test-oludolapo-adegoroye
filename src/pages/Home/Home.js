import React, { PureComponent } from 'react';

export class Home extends PureComponent {
  constructor(props) {
    super(props);
    this.state = {};
  }
  render() {
    return <div>{this.props.category}</div>;
  }
}

export default Home;
