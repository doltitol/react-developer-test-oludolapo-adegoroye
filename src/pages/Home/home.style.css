@import url("https://fonts.googleapis.com/css2?family=Raleway:wght@400;600;700&family=Roboto+Condensed:wght@700&family=Roboto:wght@400;500&family=Source+Sans+Pro&display=swap");
* {
  -webkit-box-sizing: border-box;
          box-sizing: border-box;
  padding: 0;
  margin: 0;
  scroll-behavior: smooth;
}

body {
  background: #ffffff;
}

a {
  text-decoration: none;
}

.slide-right {
  -webkit-animation: slide-right 0.5s cubic-bezier(0.25, 0.46, 0.45, 0.94) both;
  animation: slide-right 0.5s cubic-bezier(0.25, 0.46, 0.45, 0.94) both;
}

@-webkit-keyframes slide-right {
  0% {
    -webkit-transform: translateX(50px);
    transform: translateX(50px);
  }
  100% {
    -webkit-transform: translateX(0px);
    transform: translateX(0px);
  }
}

@keyframes slide-right {
  0% {
    -webkit-transform: translateX(50px);
    transform: translateX(50px);
  }
  100% {
    -webkit-transform: translateX(0px);
    transform: translateX(0px);
  }
}

.slide-down {
  -webkit-animation: slide-down 0.5s cubic-bezier(0.25, 0.46, 0.45, 0.94) both;
  animation: slide-down 0.5s cubic-bezier(0.25, 0.46, 0.45, 0.94) both;
}

@-webkit-keyframes slide-down {
  0% {
    -webkit-transform: translateY(-25%);
    transform: translateY(-25%);
  }
  100% {
    -webkit-transform: translateY(0px);
    transform: translateY(0px);
  }
}

@keyframes slide-down {
  0% {
    -webkit-transform: translateY(-25%);
    transform: translateY(-25%);
  }
  100% {
    -webkit-transform: translateY(0px);
    transform: translateY(0px);
  }
}

.home {
  background-color: #ffffff;
  padding: 80px 70px;
  width: 100%;
  margin-right: auto;
  margin-left: auto;
  margin-bottom: 80px;
}

.home-heading {
  font-family: "Raleway", sans-serif;
  font-weight: 400;
  color: #1D1F22;
  font-size: 42px;
  line-height: 67.2px;
  text-transform: capitalize;
}

.home-product {
  margin: 30px auto;
  display: -ms-grid;
  display: grid;
  -ms-grid-columns: (1fr)[3];
      grid-template-columns: repeat(3, 1fr);
  grid-gap: 12rem;
}

@media (max-width: 1400px) {
  .home-product {
    -ms-grid-columns: (1fr)[3];
        grid-template-columns: repeat(3, 1fr);
    grid-gap: 6rem;
  }
}

@media (max-width: 1200px) {
  .home-product {
    -ms-grid-columns: (1fr)[2];
        grid-template-columns: repeat(2, 1fr);
    grid-gap: 6rem;
  }
}

@media (max-width: 600px) {
  .home-product {
    -ms-grid-columns: (1fr)[1];
        grid-template-columns: repeat(1, 1fr);
    grid-gap: 6rem;
    margin: 0 10px;
  }
}

.home-loadbutton {
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
      -ms-flex-pack: center;
          justify-content: center;
  margin-top: 70px;
}
/*# sourceMappingURL=home.style.css.map */