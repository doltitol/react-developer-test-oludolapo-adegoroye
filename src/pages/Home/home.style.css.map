{
    "version": 3,
    "mappings": "ACAA,OAAO,CAAC,wKAAI;AAkBZ,AAAA,CAAC,CAAC;EACA,UAAU,EAAE,UAAU;EACtB,OAAO,EAAE,CAAC;EACV,MAAM,EAAE,CAAC;EACT,eAAe,EAAE,MAAM;CACxB;;AACD,AAAA,IAAI,CAAA;EACJ,UAAU,ECrBF,OAAO;CDsBd;;AACD,AAAA,CAAC,CAAA;EACD,eAAe,EAAE,IAAI;CACpB;;AACD,AAAA,YAAY,CAAC;EACX,iBAAiB,EAAE,WAAW,CAAC,IAAI,CAAC,oCAAoC,CAAC,IAAI;EAC7E,SAAS,EAAE,WAAW,CAAC,IAAI,CAAC,oCAAoC,CAAC,IAAI;CACtE;;AAED,kBAAkB,CAAlB,WAAkB;EAChB,EAAE;IACA,iBAAiB,EAAE,gBAAgB;IACnC,SAAS,EAAE,gBAAgB;;EAE7B,IAAI;IACF,iBAAiB,EAAE,eAAe;IAClC,SAAS,EAAE,eAAe;;;;AAI9B,UAAU,CAAV,WAAU;EACR,EAAE;IACA,iBAAiB,EAAE,gBAAgB;IACnC,SAAS,EAAE,gBAAgB;;EAE7B,IAAI;IACF,iBAAiB,EAAE,eAAe;IAClC,SAAS,EAAE,eAAe;;;;AAI9B,AAAA,WAAW,CAAC;EACV,iBAAiB,EAAE,UAAU,CAAC,IAAI,CAAC,oCAAoC,CAAC,IAAI;EAC5E,SAAS,EAAE,UAAU,CAAC,IAAI,CAAC,oCAAoC,CAAC,IAAI;CACrE;;AAED,kBAAkB,CAAlB,UAAkB;EAChB,EAAE;IACA,iBAAiB,EAAE,gBAAgB;IACnC,SAAS,EAAE,gBAAgB;;EAE7B,IAAI;IACF,iBAAiB,EAAE,eAAe;IAClC,SAAS,EAAE,eAAe;;;;AAI9B,UAAU,CAAV,UAAU;EACR,EAAE;IACA,iBAAiB,EAAE,gBAAgB;IACnC,SAAS,EAAE,gBAAgB;;EAE7B,IAAI;IACF,iBAAiB,EAAE,eAAe;IAClC,SAAS,EAAE,eAAe;;;;AD/E9B,AAAA,KAAK,CAAC;ECUJ,gBAAgB,ECPV,OAAO;EDQb,OAAO,EAAE,SAAS;EAClB,KAAK,EAAE,IAAI;EACX,YAAY,EAAE,IAAI;EAClB,WAAW,EAAE,IAAI;EACjB,aAAa,EAAE,IAAI;CD0BpB;;AAvCE,AAAD,aAAS,CAAC;EGKV,WAAW,EANC,SAAS,EAAE,UAAU;EAOjC,WAAW,EHLsB,GAAG;EGMpC,KAAK,EDRC,OAAO;ECSb,SAAS,EHPU,IAAI;EGQvB,WAAW,EHRc,MAAM;EAC7B,cAAc,EAAE,UAAU;CAC3B;;AACA,AAAD,aAAS,CAAC;EACR,MAAM,EAAE,SAAS;EACjB,OAAO,EAAE,IAAI;EACb,qBAAqB,EAAE,cAAc;EACrC,QAAQ,EAAE,KAAK;CAchB;;AAbE,MAAM,EAAE,SAAS,EAAE,MAAM;EAL3B,AAAD,aAAS,CAAC;IAMP,qBAAqB,EAAE,cAAc;IACtC,QAAQ,EAAE,IAAI;GAWf;;;AATC,MAAM,EAAE,SAAS,EAAE,MAAM;EAT1B,AAAD,aAAS,CAAC;IAUP,qBAAqB,EAAE,cAAc;IACtC,QAAQ,EAAE,IAAI;GAOf;;;AALC,MAAM,EAAE,SAAS,EAAE,KAAK;EAbzB,AAAD,aAAS,CAAC;IAcL,qBAAqB,EAAE,cAAc;IACxC,QAAQ,EAAE,IAAI;IACd,MAAM,EAAE,MAAM;GAEf;;;AACA,AAAD,gBAAY,CAAA;EACV,OAAO,EAAE,IAAI;EACb,eAAe,EAAE,MAAM;EACvB,UAAU,EAAE,IAAI;CACjB;;AACA,AAAD,aAAS,CAAA;EACT,QAAQ,EAAE,QAAQ;EAClB,UAAU,EEzBF,sBAAsB;EF0B9B,MAAM,EAAE,IAAI;EACZ,KAAK,EAAE,IAAI;EACX,IAAI,EAAE,CAAC;EACP,KAAK,EAAE,CAAC;EACR,OAAO,EAAE,CAAC;EACV,MAAM,EAAE,CAAC;EACT,OAAO,EAAE,EAAE;CACV",
    "sources": [
        "home.style.scss",
        "../../styles/index.scss",
        "../../styles/_colors.scss",
        "../../styles/_fonts.scss",
        "../../styles/_colors.scss"
    ],
    "names": [],
    "file": "home.style.css"
}