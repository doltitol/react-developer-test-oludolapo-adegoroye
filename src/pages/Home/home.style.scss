@import '../../styles/index.scss';
.home {
  @include container;
  &-heading {
    @include raleway(42px, 67.2px, 400);
    text-transform: capitalize;
    margin-top: 50px;
  }
  &-product {
    margin: 30px auto;
    display: grid;
    grid-template-columns: repeat(3, 1fr);
    grid-gap: 12rem;
     @media (max-width: 1400px){
     grid-template-columns: repeat(3, 1fr);
    grid-gap: 6rem;
    }
    @media (max-width: 1200px){
     grid-template-columns: repeat(2, 1fr);
    grid-gap: 6rem;
    }
    @media (max-width: 600px) {
       grid-template-columns: repeat(1, 1fr);
    grid-gap: 6rem;
    margin: 0 10px;
    }
  }
  &-loadbutton{
    display: flex;
    justify-content: center;
    margin-top: 70px;
  }
}
