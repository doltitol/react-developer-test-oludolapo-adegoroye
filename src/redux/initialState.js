export const initialState = {
  cartItems: JSON.parse(localStorage.getItem('cartItems') || '[]'),
  cartQuantity: JSON.parse(localStorage.getItem('cartItems') || '[]').reduce(
    (acc, item) => acc + item.qty,
    0
  ),
  currency: '$',
};
