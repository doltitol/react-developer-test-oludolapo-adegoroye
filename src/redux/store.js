import { createStore, combineReducers } from 'redux';
import cartReducer from './reducers/cartReducers';

const rootReducer = combineReducers({
  cart: cartReducer,
});
export const store = createStore(rootReducer);
