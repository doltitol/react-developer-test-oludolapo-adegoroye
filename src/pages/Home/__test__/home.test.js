import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import { Provider } from 'react-redux';
import { store } from '../../../redux/store';
import Home from '../Home';
import Categories from '../../../Categories.json';

describe('<Home /> spec', () => {
  it('renders the component', () => {
    const props = {
      categories: Categories.data.categories,
      category: 'tech',
    };
    const container = render(
      <Provider store={store}>
        <Home {...props} />
      </Provider>
    );
    expect(container.children).toMatchSnapshot();
  });
});
