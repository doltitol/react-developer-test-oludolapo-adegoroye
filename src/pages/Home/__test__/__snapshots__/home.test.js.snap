// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> spec renders the component 1`] = `undefined`;
